### PR TITLE
Updated remove member endpoint

### PR DESCRIPTION
**Pull request by kinglighthill**
- Modified rooms.py remove_member function. Added a None check to admin_id on line 130 because admin_id is optional
- Modified rooms_utils.py remove_room_member function. Changed the dict key from '_id' to 'id' on lines 150 and 160

### CODE REVIEW

**Review by man-of-mind — APPROVED** [review]
Well done boss

File: backend/endpoints/rooms.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ async def remove_member(
             detail="user not a member of the room",
         )
 
-    if admin_id not in room_data["room_members"]:
+    if admin_id is not None and admin_id not in room_data["room_members"]:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail="admin id specified not a member of the room",
```
Comment (paulinebanye) [inline backend/endpoints/rooms.py:133]: Okay, I just tested the endpoint. I couldn't remove a regular member from a channel with or without an admin_id. this is what's returned.
{
  "detail": "\"unable to remove room member\""
}

**Review by paulinebanye — CHANGES_REQUESTED** [review]
hi @kinglighthill I left a comment

**Review by mikenrowland — APPROVED** [review]
well done

File: backend/utils/room_utils.py [hunk 2]
```diff
@@ -157,5 +157,5 @@ async def remove_room_member(org_id: str, room_data: dict, member_id: str) -> di
 
     response = update_room["data"]
     response["member_id"] = member_id
-    response["room_id"] = room_data["_id"]
+    response["room_id"] = room_data["id"]
```
Comment (paulinebanye) [inline backend/utils/room_utils.py]: Working now. 👍🏻👍🏻

**Review by paulinebanye — APPROVED** [review]

File: backend/utils/room_utils.py [hunk 3]
```diff
@@ -157,5 +157,5 @@ async def remove_room_member(org_id: str, room_data: dict, member_id: str) -> di
 
     response = update_room["data"]
```
Comment (zxenonx) [inline backend/utils/room_utils.py]: We can remove matched & modified documents from the response:

```json
{
  "status": "success",
  "message": "user removed from room successfully",
  "data": {
    "matched_documents": 1,
    "modified_documents": 1,
    "member_id": "61e59f1b65934b58b8e5d1cb",
    "room_id": "61e72d1265934b58b8e5d220"
  }
}
```

File: backend/endpoints/rooms.py (hunk 1 above)
Comment (zxenonx) [inline backend/endpoints/rooms.py:133]: I tested with this room (`61e72d1265934b58b8e5d220`, Channel).
The room had only a member who is the admin, `61e59f1b65934b58b8e5d1cb`.
I was able to remove the admin by passing the admin id to both **member_id** and **admin_id** parameters.
Now, the room has no members as the admin has been removed.

This shouldn't be possible, right? @kinglighthill

File: backend/utils/room_utils.py [hunk 4]
```diff
@@ -157,5 +157,5 @@ async def remove_room_member(org_id: str, room_data: dict, member_id: str) -> di
...
     response["member_id"] = member_id
-    response["room_id"] = room_data["_id"]
+    response["room_id"] = room_id
```
Comment (zxenonx) [inline backend/utils/room_utils.py]: We can also do something like this:

`response.update({"member_id": member_id, "room_id": room_id})`

**Review by zxenonx — CHANGES_REQUESTED** [review]
Good work @kinglighthill.
I left some comments.

File: backend/endpoints/rooms.py [hunk 5]
```diff
@@ -127,11 +127,17 @@ async def remove_member(
...
         )
+    
+    if admin_id is not None and member_id is admin_id:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="cannot removed yourself"
```
Comment (paulinebanye) [inline backend/endpoints/rooms.py]:
```suggestion
            detail="cannot remove yourself"
```

**Review by paulinebanye — APPROVED** [review]
well done @kinglighthill. Channel owner is restricted from leaving channel, thus room cannot be empty.

File: backend/endpoints/rooms.py [hunk 6]
```diff
@@ -127,11 +127,17 @@ async def remove_member(
...
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail="admin id specified not a member of the room",
         )
+    
+    if admin_id is not None and member_id == admin_id:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="cannot remove yourself"
+        )
```
Comment (paulinebanye) [inline backend/endpoints/rooms.py:140]: lines 136 - 140 is completely unnecessary IMO. The endpoint already does what its supposed to without it..

**Review by zxenonx — APPROVED** [review]
Thanks @kinglighthill for the changes.

**Pull request merged**